src/net_processing.cpp

```diff
@@ -4395,45 +4395,39 @@ void PeerManagerImpl::MaybeSendAddr(CNode& pto)
         pto.m_next_local_addr_send = PoissonNextSend(current_time, AVG_LOCAL_ADDRESS_BROADCAST_INTERVAL);
     }
 
-    //
-    // Message: addr
-    //
-    if (pto.RelayAddrsWithConn() && pto.m_next_addr_send < current_time) {
-        pto.m_next_addr_send = PoissonNextSend(current_time, AVG_ADDRESS_BROADCAST_INTERVAL);
-        std::vector<CAddress> vAddr;
-        vAddr.reserve(pto.vAddrToSend.size());
-        assert(pto.m_addr_known);
-
-        const char* msg_type;
-        int make_flags;
-        if (pto.m_wants_addrv2) {
-            msg_type = NetMsgType::ADDRV2;
-            make_flags = ADDRV2_FORMAT;
-        } else {
-            msg_type = NetMsgType::ADDR;
-            make_flags = 0;
-        }
+    // We sent an `addr` message to this peer recently. Nothing more to do.
+    if (current_time <= pto.m_next_addr_send) return;
+    pto.m_next_addr_send = PoissonNextSend(current_time, AVG_ADDRESS_BROADCAST_INTERVAL);
 
-        for (const CAddress& addr : pto.vAddrToSend)
-        {
-            if (!pto.m_addr_known->contains(addr.GetKey()))
-            {
-                pto.m_addr_known->insert(addr.GetKey());
-                vAddr.push_back(addr);
-                // receiver rejects addr messages larger than MAX_ADDR_TO_SEND
-                if (vAddr.size() >= MAX_ADDR_TO_SEND)
-                {
-                    m_connman.PushMessage(&pto, msgMaker.Make(make_flags, msg_type, vAddr));
-                    vAddr.clear();
-                }
-            }
-        }
-        pto.vAddrToSend.clear();
-        if (!vAddr.empty())
-            m_connman.PushMessage(&pto, msgMaker.Make(make_flags, msg_type, vAddr));
-        // we only send the big addr message once
-        if (pto.vAddrToSend.capacity() > 40)
-            pto.vAddrToSend.shrink_to_fit();
+    assert(pto.m_addr_known);
```

Comment:
Should move this to directly after the `RelayAddrsWithConn` check -- that's what makes it true, and you rely on `m_addr_known` being not null in the `!IBD` branch above as well.

Reply:
Done